src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -397,25 +397,24 @@ def _execute_command(kwargs):
             op = get_op_handler(operation)
             try:
                 result = op(client, **kwargs) if client else op(**kwargs)
+                if no_wait_param and kwargs.get(no_wait_param, None):
+                    return None  # return None for 'no-wait'
+
+                # apply results transform if specified
+                if transform_result:
+                    return transform_result(result)
+
+                # otherwise handle based on return type of results
+                if _is_poller(result):
+                    return LongRunningOperation('Starting {}'.format(name))(result)
```

Comment:
Before "exception_handler" was introduced, I think we had logic in here specifically for AppService. Does this PR pull that logic out and transition it to the exception_handler?